spec/controllers/rooms_controller_spec.rb

```diff
@@ -0,0 +1,45 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe Api::V1::RoomsController, type: :controller do
+
+  let(:user) { create(:user)}
+  before do
+    request.headers['ACCEPT'] = 'application/json'
+    session[:user_id] = user.id
+  end
+
+  # TODO: Hadi - Make test work with current_user
+  describe 'index' do
+    it 'ids of rooms in response are matching room ids that belong to current_user' do
+      rooms = create_list(:room, 5, user: user)
+      get :index
+      expect(response).to have_http_status(:ok)
+      response_room_ids = JSON.parse(response.body)['data'].map { |room| room['id']}
+      expect(response_room_ids).to eq(rooms.pluck(:id))
+    end
+
```

Comment:
Isn't it better if we use the [request spec API ](https://relishapp.com/rspec/rspec-rails/v/5-0/docs/request-specs/request-spec) instead of the controller spec API?